ext/updater.go

```diff
@@ -352,27 +353,34 @@ func (u *Updater) StartServer(opts WebhookOpts) error {
 	}
 
 	var tls bool
-	if opts.CertFile == "" && opts.KeyFile == "" {
+	switch {
+	case opts.CertFile == "" && opts.KeyFile == "":
 		tls = false
-	} else if opts.CertFile != "" && opts.KeyFile != "" {
+	case opts.CertFile != "" && opts.KeyFile != "":
 		tls = true
-	} else {
+	default:
 		return ErrMissingCertOrKeyFile
 	}
 
+	opts.setDefaults()
+
+	ln, err := net.Listen(opts.ListenNet, opts.ListenAddr)
```

Comment:
if ListenNet or/and ListenAddr are not properly formatted, net.Listen will return a complex error message indicating what part it doesn't like (those two strings get a lot of parsing in the std lib anyway, so we should delegate this part to them).
But if you want, we could add `errors.Is()` checks on wrapped net.UnknownNetworkError, net.InvalidAddrError, net.DNSError  and net.AddrError and show a message like "something is wrong with your address/network, check your configuration".